spec/registry.md

```diff
@@ -417,6 +477,26 @@ All other values produce a _Bad Operand_ error.
 > or the type `com.ibm.icu.util.CurrencyAmount` can be used to set the currency and related
 > options (such as the number of fraction digits).
 
+### Currency Operands
+
+The _operand_ of the `:currency` function is an implementation-defined type that
+contains a number and a currency code
+or a map whose keys are `value` and `currency`.
+When the _operand_ is a map, the value of `value` MUST be a [Number Operand](#number-operands).
+When the _operand_ is a map, the value of `currency` MUST be either a 
+valid [Unicode Currency Identifier](https://cldr-smoke.unicode.org/spec/main/ldml/tr35.html#UnicodeCurrencyIdentifier)
+or an implementation-defined currency type.
```

Comment:
We should not require support for such a map value, but leave it to implementations.
```suggestion
The _operand_ of the `:currency` function is an implementation-defined type that
contains a numerical `value` and a `currency` code.
The value of `currency` MUST be either a 
valid [Unicode Currency Identifier](https://cldr-smoke.unicode.org/spec/main/ldml/tr35.html#UnicodeCurrencyIdentifier)
or an implementation-defined currency type.
```

Reply:
Fair enough. I'm going to merge your suggestion and then modify is slightly to allow multiple types (my--and thus your--current text implies only one type is allowed, which is not the intention)